libs/coin-modules/coin-sui/src/network/sdk.ts

```diff
@@ -230,33 +228,81 @@ export const executeTransactionBlock = async (params: ExecuteTransactionBlockPar
     return api.executeTransactionBlock(params);
   });
 
-// load from curos point or from begining until we reach the end
-const loadOperation = async (params: {
+/**
+ * Fetch operations for a specific address and type until the limit is reached
+ */
+export const loadOperations = async (params: {
   api: SuiClient;
-  type: OperationType;
   addr: string;
+  type: OperationType;
   cursor?: string | null | undefined;
 }): Promise<PaginatedTransactionResponse["data"]> => {
+  const operations: PaginatedTransactionResponse["data"] = [];
+  let currentCursor = params.cursor;
+
+  while (operations.length < TRANSACTIONS_LIMIT) {
+    try {
+      const { data, nextCursor, hasNextPage } = await queryTransactions({
+        ...params,
+        cursor: currentCursor,
+      });
+
+      operations.push(...data);
+
+      // If we got fewer results than the query limit or no more data, we've reached the end
+      if (data.length < TRANSACTIONS_LIMIT_PER_QUERY || !hasNextPage) {
+        break;
+      }
+
+      currentCursor = nextCursor;
+    } catch (error: unknown) {
+      if (
+        error !== null &&
+        typeof error === "object" &&
+        "type" in error &&
+        error.type === "InvalidParams"
+      ) {
+        log("coin:sui", "(network/sdk): loadOperations failed with cursor, retrying without it", {
+          error,
+          params,
+        });
+
+        currentCursor = null;
+
+        continue;
+      } else {
+        log("coin:sui", "(network/sdk): loadOperations error", { error, params });
+
+        break;
+      }
+    }
+  }
+
+  return operations;
+};
+
+/**
+ * Query transactions for given address from RPC
+ */
+export const queryTransactions = async (params: {
+  api: SuiClient;
+  addr: string;
+  type: OperationType;
+  cursor?: string | null | undefined;
+}): Promise<PaginatedTransactionResponse> => {
   const { api, addr, type, cursor } = params;
   const filter: QueryTransactionBlocksParams["filter"] =
     type === "IN" ? { ToAddress: addr } : { FromAddress: addr };
 
-  const { data, nextCursor, hasNextPage } = await api.queryTransactionBlocks({
+  return await api.queryTransactionBlocks({
     filter,
     cursor,
-    order: "ascending",
+    order: "descending",
     options: {
       showInput: true,
       showBalanceChanges: true,
       showEffects: true, // To get transaction status and gas fee details
     },
-    limit: TRANSACTIONS_REQUEST_LIMIT,
+    limit: TRANSACTIONS_LIMIT_PER_QUERY,
```

Comment:
this `limit` is regarding the total number of transactions in a call, and not the total amount of transactions you want right? ex: user has 500 transactions, you just want 200 and make each call only return 50 at the time right?

Reply:
yes right, this limit conditioned by rpc (50 is a maximum per request)